pkg/sql/catalog/typedesc/type_desc.go

```diff
@@ -657,6 +669,17 @@ func (desc *immutable) ValidateForwardReferences(
 			vea.Report(errors.AssertionFailedf("aliased type %q (%d) is dropped", typ.GetName(), typ.GetID()))
 		}
 	}
+
+	if desc.GetKind() == descpb.TypeDescriptor_COMPOSITE {
+		if ids, err := desc.GetIDClosure(); err != nil {
```

Comment:
This is almost correct. You don't actually want the closure here, only the immediate direct references. It's the established pattern because cross-reference validations can become quite expensive quite quickly.